packages/webapp/src/stories/Table/TableV2.stories.jsx

```diff
@@ -266,3 +217,64 @@ export const TasksLabourWithPagination = {
     showPagination: true,
   },
 };
+
+export const withCheckboxes = {
+  args: {
+    kind: TableKind.V2,
+    columns: getCropSalesColumns(false),
+    minRows: 10,
+    shouldFixTableLayout: true,
+    handleSelectAllClick: () => console.log('all checked!'),
+  },
+  render: (props) => {
+    const [selectedIds, setSelectedIds] = useState([]);
+    const data = getCropSalesData(10);
+    const onCheck = (e, { id }) => {
+      setSelectedIds((prevSelectedTypeIds) => {
+        const isSelected = prevSelectedTypeIds.includes(id);
+        const newSelectedIds = isSelected
+          ? selectedIds.filter((selectedId) => id !== selectedId)
+          : [...prevSelectedTypeIds, id];
+
+        return newSelectedIds;
+      });
+    };
+    const handleSelectAllClick = () => setSelectedIds(data.map(({ id }) => id));
+
+    return (
+      <>
+        <TextButton
+          color="secondary"
+          sm
+          disabled={!selectedIds.length}
+          onClick={() => setSelectedIds([])}
```

Comment:
Ah sorry for not being more clear I think I was expecting `handleSelectAllClick()` to be able to handle both selecting and deselecting all as well .

Reply:
Oh! I guess that's how that checkbox works 😂 I will fix it, thank you!

Reply:
Looks great very nice!